### ISSUE TITLE
config, camelize (and more) broken?

### ISSUE THREAD

**Issue by tcurdt** [jackfranklin/gulp-load-plugins#81]
Using this just works fine:

    G = {
      util: require('gulp-util'),
      swig: require('gulp-tc-swig'),
      markdown: require('gulp-tc-markdown'),
      frontmatter: require('gulp-front-matter'),
      vartree: require('gulp-vartree'),
      less: require('gulp-less')
    },

Replacing it with

    G = require('gulp-load-plugins')({
      pattern: [ 'gulp-*', 'gulp-tc-*' ],
      config: 'package.json',
      scope: ['dependencies', 'devDependencies', 'peerDependencies'],
      replaceString: /^gulp(-|\.)/,
      camelize: false,
      lazy: true,
      rename: {}
    }),

Gives an `Error: Cannot find module 'package.json'`. When I comment out the `config` key

    G = require('gulp-load-plugins')({
      pattern: [ 'gulp-*', 'gulp-tc-*' ],
      // config: 'package.json',
      scope: ['dependencies', 'devDependencies', 'peerDependencies'],
      replaceString: /^gulp(-|\.)/,
      camelize: false,
      lazy: true,
      rename: {}
    }),

I get `TypeError: undefined is not a function` which suggests that `G.frontmatter` does not exist. Given that `camelize` is `false` it really should. So I try the rename

    rename: {
      'gulp-front-matter': 'frontmatter'
    }

which helps. For solving the very same issue with the next  plugin I also add

    rename: {
      'gulp-front-matter': 'frontmatter',
      'gulp-tc-markdown': 'markdown'
    }

but for some reason that didn't help.

I am not sure yet whether this could be related to #80 - but something seems quite wrong here.

**Comment by jackfranklin** [comment]
When camelize is false I don't think it strips out any hyphens or anything, so you should see if:

```
G['front-matter']
G['tc-markdown']
```

gives you anything. Do those work? Else it would be great to have a repo where I can reproduce.

**Comment by tcurdt** [comment]
Ah - that's a bit counter intuitive from the option name.
IMO that should be split up into two options.

But I was just trying like this:

    var G = require('gulp-load-plugins')({
      pattern: [ 'gulp-tc-*' ]
    });

    console.log(Object.getOwnPropertyNames(G).sort());

and nothing was showing. Then I realized why - I just `npm link`ed the deps.
So that is obviously that's a "doh!" on my side. Which leaves us with two problems:

1) `Error: Cannot find module 'package.json'` if `config` is specified
2) `camelize` controlling two things a once

I am not blocked by 1) and I can work around 2) with the rename for now - but I would suggest to fix both of them. Do you want separate issue for those?

**Comment by jackfranklin** [comment]
The first is a known bug I believe - needs further looking into but isn't too severe, most people don't need to change that.

What would you like the behaviour to be when `camelize` is set to `false` ? Feel free to open another issue specifically on that :+1:

**Comment by tcurdt** [comment]
opened #82
